docs/flow/modin/data_management/functions.rst

```diff
@@ -0,0 +1,51 @@
+* GroupBy functions — Evaluates GroupBy aggregation for that type of functions that can be executed via MapReduce approach. To be able to form groups engine efficiently broadcasts `by` partitions to each partition of the source frame.
+* Map functions — Apply function to each partition in parallel.
+* MapReduce functions — Function that reduces specified axis into a scalar. First applies map function to each partition in parallel, then concatenates resulted partitions along the specified axis and applies reduction function. Note that the execution engine expects that the reduction function returns a scalar.
+* Reduction functions — Function that reduces specified axis into a scalar, but requires knowledge about the whole axis. Be aware that providing this knowledge may be expensive because the execution engine has to concatenate partitions along the specified axis. Also, note that the execution engine expects that the reduction function returns a scalar.
+* Default-to-pandas functions — Do fallback to pandas for passed function.
+
+Each template represented with a class with the corresponding name and implements ``register`` method, which takes functions to apply in an appropriate way and instantiate the related template. Functions that are passed to the ``register`` will be executed under deserialized and preprocessed (depends on the template) partitions, so the function would take one of the pandas object: ``pandas.DataFrame``, ``pandas.Series`` or ``pandas.DataFrameGroupbyObject``.
+
+.. note:: Currently, functions that are built in that way are supported only in a pandas backend (can be used only in `PandasQueryCompiler`).
+
+How to register your own function
+'''''''''''''''''''''''''''''''''
+Let's examine the actual example of how to use the function module in a sense of adding new functions.
+
+Imagine you have a complex aggregation that can be implemented into a single query but doesn't have any implementation in pandas API. If you know how to implement this aggregation efficiently in a distributed frame, you may want to use one of the described patterns. 
+
+Let's implement a function that counts non-NA values for each column or row (`pandas.DataFrame.count`). First, we need to determine the function type. MapReduce approach would be great: in a map phase, we'll count non-NA cells in each partition in parallel and then just sum its results in the reduce phase.
+
+To define the MapReduce function that does `count` + `sum` we just need to register the appropriate functions and then assign the result to the picked `QueryCompiler` (`PandasQueryCompiler` in our case): 
+.. code-block:: python
+
+    PandasQueryCompiler.custom_count = MapReduceFunction.register(pandas.DataFrame.count, pandas.DataFrame.sum)
+
+Then, we want to handle it from the DataFrame, so we need to create a way to do that:
+.. code-block:: python
+
```

Comment:
Maybe add "import modin.pandas as pd" here?

Reply:
Added import statements to the example